cmd/geth/consolecmd_test.go

```diff
@@ -15,3 +15,138 @@
+		ipc = `\\.\pipe\geth` + strconv.Itoa(rand.Int())
+	} else {
+		ws := tmpdir(t)
+		defer os.RemoveAll(ws)
+
+		ipc = filepath.Join(ws, "geth.ipc")
+	}
+	// Run the parent geth and attach with a child console
+	geth := runGeth(t, "--nat", "none", "--nodiscover", "--etherbase", coinbase, "-shh", "--ipcpath", ipc)
+	defer geth.interrupt()
+
+	time.Sleep(2 * time.Second) // Simple way to wait for the RPC endpoint to open
+	testAttachWelcome(t, geth, "ipc:"+ipc)
+}
+
+func TestHTTPAttachWelcome(t *testing.T) {
+	coinbase := "0x8605cdbbdb6d264aa742e77020dcbc58fcdce182"
+	port := strconv.Itoa(rand.Intn(65535-1024) + 1024) // Yeah, sometimes this will fail, sorry :P
+
+	geth := runGeth(t, "--nat", "none", "--nodiscover", "--etherbase", coinbase, "--rpc", "--rpcport", port)
+	defer geth.interrupt()
+
+	time.Sleep(2 * time.Second) // Simple way to wait for the RPC endpoint to open
+	testAttachWelcome(t, geth, "http://localhost:"+port)
+}
+
+func TestWSAttachWelcome(t *testing.T) {
+	coinbase := "0x8605cdbbdb6d264aa742e77020dcbc58fcdce182"
+	port := strconv.Itoa(rand.Intn(65535-1024) + 1024) // Yeah, sometimes this will fail, sorry :P
+
+	geth := runGeth(t, "--nat", "none", "--nodiscover", "--etherbase", coinbase, "--ws", "--wsport", port)
+	defer geth.interrupt()
+
+	time.Sleep(2 * time.Second) // Simple way to wait for the RPC endpoint to open
+	testAttachWelcome(t, geth, "ws://localhost:"+port)
+}
+
+func testAttachWelcome(t *testing.T, geth *testgeth, endpoint string) {
+	// Attach to a running geth note and terminate immediately
+	attach := runGeth(t, "attach", endpoint)
+	defer attach.expectExit()
+	attach.stdin.Close()
+
+	// Gather all the infos the welcome message needs to contain
+	attach.setTemplateFunc("goos", func() string { return runtime.GOOS })
+	attach.setTemplateFunc("gover", runtime.Version)
+	attach.setTemplateFunc("gethver", func() string { return verString })
+	attach.setTemplateFunc("etherbase", func() string { return geth.Etherbase })
+	attach.setTemplateFunc("niltime", func() string { return time.Unix(0, 0).Format(time.RFC1123) })
+	attach.setTemplateFunc("ipc", func() bool { return strings.HasPrefix(endpoint, "ipc") })
+	attach.setTemplateFunc("datadir", func() string { return geth.Datadir })
+	attach.setTemplateFunc("apis", func() []string {
+		var apis []string
+		if strings.HasPrefix(endpoint, "ipc") {
+			apis = append(strings.Split(rpc.DefaultIPCApis, ","), rpc.MetadataApi)
+		} else {
+			apis = append(strings.Split(rpc.DefaultHTTPApis, ","), rpc.MetadataApi)
+		}
+		sort.Strings(apis)
+		return apis
+	})
+	attach.setTemplateFunc("prompt", func() string { return console.DefaultPrompt })
+
+	// Verify the actual welcome message to the required template
+	attach.expect(`
+Welcome to the Geth JavaScript console!
+
+instance: Geth/v{{gethver}}/{{goos}}/{{gover}}
+coinbase: {{etherbase}}
+at block: 0 ({{niltime}}){{if ipc}}
+ datadir: {{datadir}}{{end}}
+ modules:{{range apis}} {{.}}:1.0{{end}}
+
+{{prompt}}
```

Comment:
Not sure about that tbh. Imho it provides more information to the user about the current status of the chain than having a random timezone you need to calculate with.